cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
I guess there are two aspects to this, i) using up extra space by creating things that may never be used and ii) the time of execution to do so.
Regarding i) when we talked about this @tomblench mentioned it could be potentially confusing in debugging if this stuff was automatically created even if indexing wasn't being used. I'm sure we can deal with that, and if we're talking a few bytes vs a few MB then it probably isn't worth the hassle of deferring.
Regarding ii) it could change the startup profile for apps that may currently intentionally defer the `IndexManager` creation to some later stage. I guess it depends enormously on how long it actually takes to do this stuff.

You could add an init and boolean check, but it would be annoying to remember to add the init check for every query method call unless we integrated it into a superclass of all the necessary callables to keep it in one place for example.

Another option would be if DatastoreImpl was reverted to non-abstract then it could have a private method for getQueryableDatastore that could do the init check and initiatilization and then the delegated method calls could be `getQueryableDatastore().find` etc.